src/colmap/sfm/observation_manager_test.cc

```diff
@@ -85,39 +85,47 @@ TEST(ObservationManager, FilterPoints3D) {
   const point3D_t point3D_id1 =
       reconstruction.AddPoint3D(Eigen::Vector3d::Random(), Track());
   EXPECT_EQ(reconstruction.NumPoints3D(), 1);
-  obs_manager.FilterPoints3D(0.0, 0.0, std::unordered_set<point3D_t>{});
+  EXPECT_EQ(
+      obs_manager.FilterPoints3D(0.0, 0.0, std::unordered_set<point3D_t>{}), 0);
   EXPECT_EQ(reconstruction.NumPoints3D(), 1);
-  obs_manager.FilterPoints3D(
-      0.0, 0.0, std::unordered_set<point3D_t>{point3D_id1 + 1});
+  EXPECT_EQ(obs_manager.FilterPoints3D(
+                0.0, 0.0, std::unordered_set<point3D_t>{point3D_id1 + 1}),
+            0);
   EXPECT_EQ(reconstruction.NumPoints3D(), 1);
-  obs_manager.FilterPoints3D(
-      0.0, 0.0, std::unordered_set<point3D_t>{point3D_id1});
+  EXPECT_EQ(obs_manager.FilterPoints3D(
```

Comment:
This is odd. Wouldn't we expect this to be non-zero?

Reply:
Ah, no, because the point has an empty track. I suggest to add observations, so we can make the expectation more meaningful? Same for other instances below?